src/state/migrations.ts

```diff
@@ -0,0 +1,69 @@
+import { createMigrate, MigrationManifest, PersistedState, PersistMigrate } from 'redux-persist'
+import { MigrationConfig } from 'redux-persist/es/createMigrate'
+
+import { legacyTransactionMigrations, legacyUserMigrations } from './legacyMigrations'
+import { initialState as initialListsState } from './lists/reducer'
+import { initialState as initialTransactionsState } from './transactions/reducer'
+import { initialState as initialUserState } from './user/reducer'
+
+function tryParseOldState<T>(value: string | null, fallback: T): T {
+  try {
+    return value ? JSON.parse(value) : fallback
+  } catch (e) {
+    return fallback
+  }
+}
+
+/**
+ * These run once per state re-hydration when a version mismatch is detected.
+ * Keep them as lightweight as possible.
+ *
+ * Migration functions should not assume that any value exists in localStorage previously,
+ * because a user may be visiting the site for the first time or have cleared their localStorage.
+ */
+
+// The target version number is the key
+export const migrations: MigrationManifest = {
+  0: () => {
+    const oldTransactions = localStorage.getItem('redux_localstorage_simple_transactions')
```

Comment:
Can you move the legacy migration code into `src/state/migrations/legacy.ts`.
Can you move the `0` (a no-op) migration into `src/state/migrations/0.ts`.
This will prevent this file from growing over time, while still keeping migrations organized.